share/eds/store.go

```diff
@@ -69,13 +71,20 @@ func NewStore(basepath string, ds datastore.Batching) (*Store, error) {
 		return nil, fmt.Errorf("failed to create DAGStore: %w", err)
 	}
 
-	return &Store{
+	store := &Store{
 		basepath: basepath,
 		dgstr:    dagStore,
 		topIdx:   invertedRepo,
 		carIdx:   fsRepo,
 		mounts:   r,
-	}, nil
+	}
+
+	store.bs, err = NewEDSBlockstore(store)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create EDSBlockstore: %w", err)
+	}
```

Comment:
optional nit: creat the blockstore first and then the store

Reply:
But the blockstore requires the store for the backreference, or am I missing something?

Reply:
It's a private structure so that you can set up the Store afterward, but that's a very minor thing. Not worth spending any more cycles on it

Reply:
The way it is right now looks very good, so resolved